qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,532 @@
+    a_rs = edge_operator_aij(edge_list, r, s)
+    a_pq = -a_pq if q < p else a_pq
+    a_rs = -a_rs if s < r else a_rs
+
+    id_op = _pauli_id(edge_list.shape[1])
+    qubit_op = (a_pq * a_rs) * (
+        -id_op
+        - b_p * b_q
+        + b_p * b_r
+        + b_p * b_s
+        + b_q * b_r
+        + b_q * b_s
+        - b_r * b_s
+        + b_p * b_q * b_r * b_s  ## Agrees with SW2018 eq 37 and OpenFermion. Aqua had `-`.
+    )
+    final_coeff = 0.125
+    qubit_op = (final_coeff * h2_pqrs) * qubit_op
+    return qubit_op
+
+
+def _number_excitation(  # pylint: disable=invalid-name
+    edge_list: np.ndarray, p: int, q: int, r: int, s: int, h2_pqrs: float
+) -> SparsePauliOp:
+    b_p = edge_operator_bi(edge_list, p)
+    b_q = edge_operator_bi(edge_list, q)
+    id_op = _pauli_id(edge_list.shape[1])
+    if p == r:
+        b_s = edge_operator_bi(edge_list, s)
+        a_qs = edge_operator_aij(edge_list, q, s)
+        a_qs = -a_qs if s < q else a_qs
+        qubit_op = (a_qs * b_s + b_q * a_qs) * (id_op - b_p)
+        final_coeff = 1j * 0.25
+    elif p == s:
+        b_r = edge_operator_bi(edge_list, r)
+        a_qr = edge_operator_aij(edge_list, q, r)
+        a_qr = -a_qr if r < q else a_qr
+        qubit_op = (a_qr * b_r + b_q * a_qr) * (id_op - b_p)
+        final_coeff = 1j * -0.25
+    elif q == r:
+        b_s = edge_operator_bi(edge_list, s)
+        a_ps = edge_operator_aij(edge_list, p, s)
+        a_ps = -a_ps if s < p else a_ps
+        qubit_op = (a_ps * b_s + b_p * a_ps) * (id_op - b_q)
+        final_coeff = 1j * -0.25
+    elif q == s:
+        b_r = edge_operator_bi(edge_list, r)
+        a_pr = edge_operator_aij(edge_list, p, r)
+        a_pr = -a_pr if r < p else a_pr
+        qubit_op = (a_pr * b_r + b_p * a_pr) * (id_op - b_q)
+        final_coeff = 1j * 0.25
+    else:
+        raise ValueError(f"unexpected sequence of indices: {p}, {q}, {r}, {s}")
+    qubit_op = (final_coeff * h2_pqrs) * qubit_op
+    return qubit_op
+
+
+def _unpack_term(
+    term_str: str, expand_number_op: bool = False
+) -> Tuple[Tuple[int, int, int], List[Tuple[int, str]]]:
+    """
+    Return a tuple specifying the counts of kinds of operators in `term_str` and
+    a list of the factors and their indices in `term_str`.
+
+    The factors are represented by tuples of the form `(i, c)`, where `i` is an index
+    and `c` is a character.
+    Allowed characters in `term_str` are 'N+-I`.
+    The returned tuple contains counts for `N`, `+`, and `-`, in that order. Identity operators
+    are ignored.
+
+    Args:
+       `term_str`: a string of characters in `+-NI`.
+       `expand_number_op`: if `True`, number operators are expanded to `(i, '+')`, `(i, '-')`
+         in the returned list of factors.
+
+    Returns:
+       tuple: A tuple of two elements. First, a tuple of three integers giving the number of
+       number-, raising-, and -lowering operators. Second a list of factors represented by
```

Comment:
consistency with below
```suggestion
       number, raising, and lowering operators. Second a list of factors represented by
```